Source/AST/AST.swift

```diff
@@ -364,30 +364,35 @@ extension Inline : Renderable {
             
         case .link(let children, title: _, let urlStr):
             let content = children.render(with: style)
-            
-            
-            let detector = try? NSDataDetector(types: NSTextCheckingResult.CheckingType.link.rawValue)
-            
-            // validate links
-            if let urlStr = urlStr, let detector = detector {
-                
-                let getURL: (String) -> URL? = { urlStr in
-                    let match = detector.firstMatch(in: urlStr, options: [], range: NSMakeRange(0, (urlStr as NSString).length))
-                    return match?.url
-                }
-                
-                // first ensure the urlStr is valid
-                if let url = getURL(urlStr), Application.shared.canOpenURL(url) {
-                    // overwrite styling to avoid bold, italic, code links
-                    content.addAttributes(style.defaultAttributes)
-                    content.addAttribute(.markdown, value: Markdown.link, range: content.wholeRange)
-                    content.addAttribute(.link, value: url, range: content.wholeRange)
-                    return content
+
+            let styleBlock: (URL) -> Void = {
```

Comment:
This is just my personal opinion: I usually prefer to do this kind of styling action in a separate method in stead of having a block defined here inside. In this case, I would create an extension of URL and define a method which apply the markdown link style to the URL.

IMHO: defining all the block there could make a little verbose and long the method.

Reply:
Couldn't really extract this as an extension because it captures values. But I extracted it to a method instead. dcc5516352a300fc626334b41d04e1f3034a1f8d